os/mm/umm_heap/umm_realloc.c

```diff
@@ -89,9 +89,7 @@ void *realloc_at(int heap_index, void *oldmem, size_t size)
 {
 	void *ret;
 #ifdef CONFIG_DEBUG_MM_HEAPINFO
-	mmaddress_t caller_retaddr = 0;
-
-	ARCH_GET_RET_ADDRESS(caller_retaddr)
+	mmaddress_t caller_retaddr = (mmaddress_t)__builtin_return_address(0);
```

Comment:
We need to abstract this to at least some extent.

While it is supported by GCC and Clang, it may not be supported by certain other compilers (e.g., MSVC).

Looking at the usage of __builtin_return_address(0) in NuttX and Linux, we can observe that it is abstracted using a #define directive.
https://github.com/apache/nuttx/blob/5594c2e8875730f2f65b9fab57806e4c3c046379/include/nuttx/compiler.h#L171
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/linux/instruction_pointer.h#n7

To ensure future scalability and compatibility, it is necessary to abstract this functionality like below
```
#define GET_RETURN_ADDRESS   (mmaddress_t)__builtin_return_address(0);
```